server/common/annotations.py

```diff
@@ -298,20 +337,31 @@ def write_labels(self, df, data_adaptor):
             dataset_id = uuid.uuid4()
             dataset = CellxGeneDataset(id=dataset_id, name=dataset_name)
             self.db.session.add(dataset)
+            self.db.session.commit()
 
-        uri = f"{self.directory_path}/{dataset_name}/{uid}/{timestamp}"
+        uri = f"{self.directory_path}/{dataset_name}/{self.user_id}/{timestamp}"
         if "s3" in uri:
             pass
         else:
             os.makedirs(uri, exist_ok=True)
-        schema_hints = {}
+        schema_hints, values = generate_schema_hints_and_convert_value_types(df)
+
         annotation = Annotation(
             tiledb_uri=uri,
-            user_id=uid,
+            user_id=self.user_id,
             dataset_id=str(dataset_id),
             schema_hints=json.dumps(schema_hints)
         )
-        # todo in future pr -- write df to tiledb, store at uri
+        with self.label_lock:
+            if not df.empty:
+                self.check_category_names(df)
+                # convert to tiledb datatypes
+                for col in df:
+                    df[col] = df[col].astype(cxg_dtype(df[col]))
+                try:
+                    tiledb.from_pandas(uri, df)
+                except Exception as e:
+                    print(e)
```

Comment:
Don't ignore the exception.  Raise an exception here, which will cause the error to be logged, and the api will return a non-success code.

Reply:
thats was leftover from testing. removed